dango/lending/src/core.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::BTreeMap;
+
+use grug::{Coin, Inner, Number, Udec128};
+
+use crate::MARKETS;
+
+use {
+    anyhow::bail,
+    dango_types::lending::{NAMESPACE, SUBNAMESPACE},
+};
+
+use {
+    dango_types::lending::Market,
+    grug::{Coins, Denom, QuerierWrapper, Storage, Timestamp},
+};
+
+/// Calculates the amount of LP tokens to mint for a deposit.
+/// Returns the amount of LP tokens and the updated markets.
+pub fn calculate_deposit(
+    storage: &dyn Storage,
+    querier: &QuerierWrapper,
```

Comment:
Actually, `QuerierWrapper` is just a `&dyn Querier`, so we can derive `Copy` on it, so you can just use `QuerierWrapper` without taking the reference. Let me do this in a separate PR.